src/shogun/base/SGObject.h

```diff
@@ -1011,5 +1021,64 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+#ifndef SWIG
+template <typename T1, typename T2>
+bool execute_dispatch_array_type(const CSGObject* obj, const std::string& name, T2&& lambda)
+{
+	Tag<CDynamicObjectArray*> tag_array_sg(name);
+	if (obj->has(tag_array_sg))
+	{
+		auto dispatched = obj->get(tag_array_sg);
+		lambda(*dispatched); // is stored as a pointer
+		return true;
+	}
+
+	Tag<std::vector<T1*>> tag_vector(name);
+	if (obj->has(tag_vector))
+	{
+		auto dispatched = obj->get(tag_vector);
+		lambda(dispatched);
+		return true;
+	}
+
+	return false;
+}
+
+struct ByName
+{
+};
+
+struct ByNameIndex
+{
+	ByNameIndex(index_t index) : m_index(index) {}
+	index_t m_index;
+};
+
+template <typename T>
+CSGObject* get(const CSGObject* obj, const std::string& name, ByName)
+{
+	return obj->get<T*>(name);
+}
+
+template <typename T>
+CSGObject* get(const CSGObject* obj, const std::string& name, ByNameIndex i)
+{
+	return obj->get<T>(name, i.m_index);
+}
+
+template<typename T>
+CSGObject* get_dispatch_all_base_types(const CSGObject* obj, const std::string& name,
+		T&& how)
+{
+	if (auto* result = get<CKernel>(obj, name, how))
```

Comment:
this is a single central list of trying different get versions for all base types